OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectAtom.cs

```diff
@@ -36,94 +35,119 @@ class DreamMetaObjectAtom : DreamMetaObjectDatum {
-            } else if (variableName == "dir") {
-                UpdateAppearance(dreamObject, appearance => {
-                    appearance.Direction = (AtomDirection)variableValue.GetValueAsInteger();
-                });
-            } else if (variableName == "transform") {
-                DreamObject matrix = variableValue.GetValueAsDreamObjectOfType(DreamPath.Matrix);
-
-                UpdateAppearance(dreamObject, appearance => {
-                    appearance.Transform[0] = matrix.GetVariable("a").GetValueAsFloat();
-                    appearance.Transform[1] = matrix.GetVariable("d").GetValueAsFloat();
-                    appearance.Transform[2] = matrix.GetVariable("b").GetValueAsFloat();
-                    appearance.Transform[3] = matrix.GetVariable("e").GetValueAsFloat();
-                    appearance.Transform[4] = matrix.GetVariable("c").GetValueAsFloat();
-                    appearance.Transform[5] = matrix.GetVariable("f").GetValueAsFloat();
-                });
-            } else if (variableName == "overlays") {
-                if (oldVariableValue.TryGetValueAsDreamList(out DreamList oldList)) {
-                    oldList.Cut();
-                    oldList.ValueAssigned -= OverlayValueAssigned;
-                    oldList.BeforeValueRemoved -= OverlayBeforeValueRemoved;
-                    _atomManager.OverlaysListToAtom.Remove(oldList);
+            switch (variableName)
+            {
+                case "icon":
+                    UpdateAppearance(dreamObject, appearance => {
+                        if (variableValue.TryGetValueAsDreamResource(out DreamResource resource)) {
+                            appearance.Icon = resource.ResourcePath;
+                        } else {
+                            appearance.Icon = null;
+                        }
+                    });
+                    break;
+                case "icon_state":
+                    UpdateAppearance(dreamObject, appearance => {
+                        variableValue.TryGetValueAsString(out appearance.IconState);
+                    });
+                    break;
+                case "pixel_x":
+                    UpdateAppearance(dreamObject, appearance => {
+                        appearance.PixelOffset.X = variableValue.GetValueAsInteger();
+                    });
+                    break;
+                case "pixel_y":
+                    UpdateAppearance(dreamObject, appearance => {
+                        appearance.PixelOffset.Y = variableValue.GetValueAsInteger();
+                    });
+                    break;
+                case "layer":
+                    UpdateAppearance(dreamObject, appearance => {
+                        appearance.Layer = variableValue.GetValueAsFloat();
+                    });
+                    break;
+                case "invisibility":
+                    var vis = variableValue.TryGetValueAsFloat(out var val) ? Convert.ToInt32(Math.Floor(val)) : 0;
```

Comment:
`TryGetValueAsInteger()` will floor a float, and sets the output value to 0 on failure
```suggestion
                    variableValue.TryGetValueAsInteger(out int vis);
```